lib/Doctrine/DBAL/Exception/ConstraintViolationException.php

```diff
@@ -28,4 +28,17 @@
  */
 class ConstraintViolationException extends ServerException
 {
+	/**
+	 * Get the name of affected column.
+	 *
+	 * @return string|NULL
+	 */
+	public function getAffectedColumnName()
+	{
+		if (preg_match_all('/^DETAIL\:\s+Key\s+\((\w+)\)/im', $this->getMessage(), $matches)) {
```

Comment:
This will not work for all vendors and drivers as error messages are different for each.